### PR TITLE
Eks functional tests implementation

### PR DESCRIPTION
**Pull request by gurevichdmitry**
- file system tests
- kubelet process tests
- psp tests
- eks configuration tests
- aws managed services tests

### CODE REVIEW

File: tests/product/tests/data/eks_test_case.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
+"""
```
[the remark this answers is not in the document]
Comment (gurevichdmitry) [reply tests/product/tests/data/eks_test_case.py:1]: I've tried it, but the problem in register_param function and it is a bit complicated due to Python limitations.

File: justfile [hunk 2]
```diff
@@ -117,7 +117,7 @@ deploy-tests-helm values_file target range='':
   helm upgrade --wait --timeout={{TIMEOUT}} --install --values {{values_file}} --set testData.marker={{target}} --set testData.range={{range}} --set elasticsearch.imageTag={{ELK_STACK_VERSION}} --set kibana.imageTag={{ELK_STACK_VERSION}} --namespace={{NAMESPACE}} {{TESTS_RELEASE}}  tests/deploy/k8s-cloudbeat-tests/
 
 purge-tests:
-	helm del {{TESTS_RELEASE}} -n {{NAMESPACE}} & kubectl delete pvc --all
+	helm del {{TESTS_RELEASE}} -n {{NAMESPACE}} & kubectl delete pvc --all -n {{NAMESPACE}}
```
[the remark this answers is not in the document]
Comment (gurevichdmitry) [reply justfile:120]: When deleting PVC need to provide namespace.

File: tests/product/tests/data/aws/managed_services_test_cases.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    'test-eks-config-1': {
+        **cis_eks_5_1_1,
+        # **cis_eks_5_4_3_config_1,
+        **cis_eks_5_4_5_config_1,
+        **dict(zip(cis_eks_2_1_1_config_1.keys(),
+                   skip_param_case(skip_list=[*cis_eks_2_1_1_config_1.values()],
+                                   data_to_report=SkipReportData(
+                                       skip_reason='This rule is implemented partially',
+                                       url_title='security-team: #3929',
+                                       url_link='https://github.com/elastic/security-team/issues/3929'
```
[the remark this answers is not in the document]
Comment (gurevichdmitry) [reply tests/product/tests/data/aws/managed_services_test_cases.py:81]: According to [this issue](https://github.com/elastic/security-team/issues/4471) the rule 2.1.1 is not working, and refers to issue provided in skip case.

File: tests/product/tests/data/aws/managed_services_test_cases.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+                                   ))))
+    },
+    'test-eks-config-2': {
+        **dict(zip(cis_eks_2_1_1_config_2.keys(),
+                   skip_param_case(skip_list=[*cis_eks_2_1_1_config_2.values()],
+                                   data_to_report=SkipReportData(
+                                       skip_reason='This rule is implemented partially',
+                                       url_title='security-team: #3929',
```
[the remark this answers is not in the document]
Comment (gurevichdmitry) [reply tests/product/tests/data/aws/managed_services_test_cases.py:89]: According to [this issue](https://github.com/elastic/security-team/issues/4471) the rule 2.1.1 is not working, and refers to issue provided in skip case.

File: tests/product/tests/data/k8s_object/eks_k8s_object_test_cases.py [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+# cis_eks_4_2_8_fail = EksKubeObjectCase(
+#     rule_tag='CIS 4.2.8',
+#     test_resource_id='eks-psp-failures',
+#     expected=RULE_FAIL_STATUS
+# )
+#
+# cis_eks_4_2_8 = {
+#     '4.2.8 PSP spec.securityContext.capabilities default == not presented eval passed': cis_eks_4_2_8_pass,
+#     '4.2.8 PSP spec.securityContext.capabilities.add==["NET_ADMIN", "SYS_TIME"] eval failed': cis_eks_4_2_8_fail
+# }
```
[the remark this answers is not in the document]
Comment (gurevichdmitry) [reply tests/product/tests/data/k8s_object/eks_k8s_object_test_cases.py]: Skip test is added and the link to task is provided.

File: tests/product/tests/test_eks_k8s_objects_rules.py [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    ("rule_tag", "test_resource_id", "expected"),
+    [
+        *eks_k8s_object_tc.cis_eks_4_2_1.values(),
+        *eks_k8s_object_tc.cis_eks_4_2_2.values(),
+        *eks_k8s_object_tc.cis_eks_4_2_3.values(),
+        *eks_k8s_object_tc.cis_eks_4_2_4.values(),
+        *eks_k8s_object_tc.cis_eks_4_2_5.values(),
+        *eks_k8s_object_tc.cis_eks_4_2_6.values(),
+        *eks_k8s_object_tc.cis_eks_4_2_7.values(),
+        # *eks_k8s_object_tc.cis_eks_4_2_8.values(), # TODO to check if finding retrieved on real env
```
[the remark this answers is not in the document]
Comment (gurevichdmitry) [reply tests/product/tests/test_eks_k8s_objects_rules.py]: On real environment finding of rule 4.2.8 exists. Skip test is added and the link to task is provided.

File: tests/product/tests/test_eks_k8s_objects_rules.py [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+     ],
+    ids=[
+        *eks_k8s_object_tc.cis_eks_4_2_1.keys(),
+        *eks_k8s_object_tc.cis_eks_4_2_2.keys(),
+        *eks_k8s_object_tc.cis_eks_4_2_3.keys(),
+        *eks_k8s_object_tc.cis_eks_4_2_4.keys(),
+        *eks_k8s_object_tc.cis_eks_4_2_5.keys(),
+        *eks_k8s_object_tc.cis_eks_4_2_6.keys(),
+        *eks_k8s_object_tc.cis_eks_4_2_7.keys(),
+        # *eks_k8s_object_tc.cis_eks_4_2_8.keys(),
```
[the remark this answers is not in the document]
Comment (gurevichdmitry) [reply tests/product/tests/test_eks_k8s_objects_rules.py]: On real environment finding of rule 4.2.8 exists. Skip test is added and the link to task is provided.

**Review by oren-zohar — APPROVED** [review]
Looks great! 🚀

**Pull request merged**